backend/session/logind.c

```diff
@@ -819,6 +847,11 @@ static struct wlr_session *logind_session_create(struct wl_display *disp) {
 		}
 	}
 
+	if (!set_type(session)) {
+		// Not fatal
+		wlr_log(WLR_INFO, "Failed to set logind session type to wayland");
```

Comment:
This message isn't strictly accurate. If it was the `setenv()` call that caused the `false` return, the logind session type *was* set to `wayland`.

Reply:
True. We already log both failure modes correctly in `set_type`, maybe this log is just redundant?

Reply:
> maybe this log is just redundant?

I'd say so, yes. Since it's just a "best effort" attempt to set the session type and not a fatal error, it seems like making the function a `void` return and just calling `set_type(session)` would be sufficient.